lib/matplotlib/axis.py

```diff
@@ -781,14 +781,36 @@ def get_children(self):
         return [self.label, self.offsetText,
                 *self.get_major_ticks(), *self.get_minor_ticks()]
 
-    def _reset_major_tick_kw(self):
+    def _reset_major_tick_kw(self, keep_tick_and_label_visibility=False):
+        """
+        Reset major tick params to defaults.
+
+        Shared subplots pre-configure tick and label visibility. To keep this
+        beyond an Axis.clear() operation, we may
```

Comment:
```suggestion
        beyond an `Axis.clear()` operation, we may
```
(Or something similar. If possible. For both methods.)